sdk/python/lib/pulumi/config.py

```diff
@@ -231,23 +266,41 @@ def _get_object(
         except Exception as e:
             raise ConfigTypeError(self.full_key(key), v, "JSON object") from e
 
-    def get_object(self, key: str) -> Optional[Any]:
+    def get_object(self, key: str, default: Optional[Any] = None) -> Optional[Any]:
         """
-        Returns an optional configuration value, as an object, by its key, or undefined if it
+        Returns an optional configuration value, as an object, by its key,
+        a default value if that key is unset and a default is provided, or undefined if it
         doesn't exist. This routine simply JSON parses and doesn't validate the shape of the
         contents.
+
+        :param str key: The requested configuration key.
+        :param Optional[Any] default: An optional fallback value to use if the given configuration key is not set.
+        :return: The configuration key's value, or None if one does not exist.
+        :rtype: Optional[Any]
+        :raises ConfigTypeError: The configuration value existed but couldn't be coerced to float.
         """
-        return self._get_object(key, self.get_secret_object, self.get_object)
+        config_candidate = self._get_object(
+            key, self.get_secret_object, self.get_object
+        )
+        return config_candidate if config_candidate is not None else default
 
-    def get_secret_object(self, key: str) -> Optional[Output[Any]]:
+    def get_secret_object(
+        self, key: str, default: Optional[Any] = None
+    ) -> Optional[Output[Any]]:
         """
-        Returns an optional configuration value, as an object, by its key, marking it as a secret or
-        undefined if it doesn't exist. This routine simply JSON parses and doesn't validate the
+        Returns an optional configuration value, as an object, by its key, marking it as a secret,
+        a default value if that key is unset and a default is provided,
+        or undefined if it doesn't exist. This routine simply JSON parses and doesn't validate the
         shape of the contents.
+
+        :param str key: The requested configuration key.
+        :param Optional[Any] default: An optional fallback value to use if the given configuration key is not set.
+        :return: The configuration key's value, or None if one does not exist.
+        :rtype: Optional[Any]
+        :raises ConfigTypeError: The configuration value existed but couldn't be coerced to float.
         """
-        v = self._get_object(key)
-        if v is None:
-            return None
+        config_candidate = self._get_object(key)
+        v = config_candidate if config_candidate is not None else default
```

Comment:
My curiosity got the better of me here:
![image](https://user-images.githubusercontent.com/4128006/184005092-1b1e8c33-1993-48c4-a952-50de19e241d6.png)
![image](https://user-images.githubusercontent.com/4128006/184005118-94dc58f2-0737-48e4-97ed-2d805a2e11e6.png)

`Optional[Any]` collapses down to `Any`, so mypy had no idea this would be an issue!